libr/util/pj.c

```diff
@@ -168,7 +207,11 @@ R_API PJ *pj_ka(PJ *j, const char *k) {
 R_API PJ *pj_ks(PJ *j, const char *k, const char *v) {
 	r_return_val_if_fail (j && k && v, j);
 	pj_k (j, k);
-	pj_s (j, v);
+	if (j->str_encoding != PJ_ENCODING_DEFAULT) {
```

Comment:
so every `pj_ks` actually get encoded? Is that done on purpose?

Reply:
It is on purpose yeah. Ultimately, there are trade-offs here. 

If we encode only some things then the user now needs to selectively decode certain things only, and that involves them writing code to check if it's encoded or not, or they need to try out all of the commands ahead of time to work out what fields need to be decoded. On the flip-side, if we encode everything, we encode things that are already safe JSON strings (per the spec) so its added work - albeit shouldn't be that noticeable.

IMO, the default behaviour (no encoding) is unchanged, so everyone is unaffected. Users will opt into encoding to fit their circumstances, and it may be better longer-term to extend this functionality to allow them to configure which fields explicitly they want to be encoded (via config file or something), but I think as the first version of this, we probably want to encode all. Plus if someone opts in for encoding, they'll know that they need to decode each thing and we can always adjust it if it's undesirable.